graphql-bal-service/src/main/java/io/ballerina/graphql/schema/utils/Utils.java

```diff
@@ -0,0 +1,340 @@
+            if (listFiles != null) {
+                schemaName = checkAvailabilityOfGivenName(schemaName, listFiles);
+            }
+        }
+        return schemaName;
+    }
+
+    /**
+     * This method for check the availability of the given file name in the output directory.
+     *
+     * @param schemaName     schema file name
+     * @param listFiles      generated files
+     *@return file name with duplicate number tag
+     */
+    private static String checkAvailabilityOfGivenName(String schemaName, File[] listFiles) {
+        for (File file : listFiles) {
+            if (System.console() != null && file.getName().equals(schemaName)) {
+                String userInput = System.console().readLine("There is already a file named '" + file.getName() +
+                        "' in the target location. Do you want to overwrite the file? [y/N] ");
+                if (!Objects.equals(userInput.toLowerCase(Locale.ENGLISH), "y")) {
+                    schemaName = setGeneratedFileName(listFiles, schemaName);
+                }
+            }
+        }
+        return schemaName;
+    }
+
+    /**
+     * This method for setting the file name for generated file.
+     *
+     * @param listFiles      generated files
+     * @param fileName       File name
+     */
+    private static String setGeneratedFileName(File[] listFiles, String fileName) {
+        int duplicateCount = 0;
+        for (File listFile : listFiles) {
+            String listFileName = listFile.getName();
+            if (listFileName.contains(".") && ((listFileName.split("\\.")).length >= 2)
+                    && (listFileName.split("\\.")[0].equals(fileName.split("\\.")[0]))) {
+                duplicateCount++;
+            }
+        }
+        return fileName.split("\\.")[0] + PERIOD + duplicateCount + PERIOD + fileName.split("\\.")[1];
+    }
+
+    /**
+     * This method use for write the generated SDL schema string.
+     *
+     * @param filePath     output file path
+     * @param content      SDL schema string
+     */
+    public static void writeFile(Path filePath, String content) throws SchemaGenerationException {
+        try (FileWriter writer = new FileWriter(filePath.toString(), StandardCharsets.UTF_8)) {
+            writer.write(content);
+        } catch (IOException e) {
+            throw new SchemaGenerationException(DiagnosticMessages.SDL_SCHEMA_103, null, e.getMessage());
+        }
+    }
+
+    /**
+     * This method create the given output directory if not exist.
+     *
+     * @param outputPath     output file path
+     */
+    public static void createOutputDirectory(Path outputPath) {
+        File outputDir = new File(outputPath.toString());
+        if (!outputDir.exists()) {
+            outputDir.mkdirs();
+        }
+    }
+
+    /**
+     * This {@code NullLocation} represents the null location allocation for scenarios which has no location.
+     */
+    public static class NullLocation implements Location {
```

Comment:
What is the reason for defining an inner class?

Reply:
Replied in https://github.com/ballerina-platform/graphql-tools/pull/118#discussion_r1036889410